### DISCUSSION TITLE
Omnisharp installation error

### DISCUSSION THREAD

**Discussion by hwhite85** [mason-org/mason.nvim#1282] [Q&A] [answered]
Hi All,

I'm trying to install omnisharp ls and keep getting "Cannot write exec wrapper to "omnisharp" because the file already exists". I've removed the existing omnisharp folder in the data directory, but this had no effect. 

Does anyone know what might be causing the issue? 

Thanks

**Comment by williamboman** [comment]
If you've manually tried to clean up things in Mason's directory it may have "corrupted" the installation if not all related artifacts were removed. Try installing again with the `--force` flag, `:MasonInstall --force`

**Comment by hwhite85** [comment]
Tried it with no luck.  It looks like it is failing on the "Registering bin links" line: 

`[DEBUG Thu May  4 12:29:12 2023] ...te/pack/packer/start/mason.nvim/lua/mason-core/fetch.lua:34: Fetching URL "https://raw.githubusercontent.com/OmniSharp/omnisharp-vscode/master/package.json"

[DEBUG Thu May  4 12:29:12 2023] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="curl", spawn_opts={
  args = { "-H", "User-Agent: mason.nvim (+https://github.com/williamboman/mason.nvim)", "-fsSL", "-X", "GET", "https://raw.githubusercontent.com/OmniSharp/omnisharp-vscode/master/package.json" }
}

[DEBUG Thu May  4 12:29:12 2023] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 45157

[DEBUG Thu May  4 12:29:13 2023] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=45157 exited with exit_code=0, signal=0

[DEBUG Thu May  4 12:29:13 2023] ...rt/mason.nvim/lua/mason-core/installer/registry/link.lua:157: Registering bin links Package(name=omnisharp) {
  omnisharp = "dotnet:OmniSharp.dll"
}

[ERROR Thu May  4 12:29:13 2023] ...acker/start/mason.nvim/lua/mason-core/installer/init.lua:200: Installation failed for Package(name=omnisharp) error='Cannot write exec wrapper to "omnisharp" because the file already exists.'
`

**Reply by williamboman** [reply, marked as answer]
Hm there seems to be [an issue](https://github.com/mason-org/mason-registry/actions/runs/4886142730) with the package itself on case-insensitive file systems actually. https://github.com/mason-org/mason-registry/pull/1478 should fix this, (once merged run `:MasonUpdate` and try installing again)

**Reply by hwhite85** [reply]
That did it. Thanks!
